src/handlers/strategies/serveFile.ts

```diff
@@ -33,7 +33,13 @@ function getStatusCode(request: Request, objectHasBody: boolean): number {
     return 412;
   }
 
-  // We weren't given a body
+  // We weren't given a body and preconditions succeeded
+
+  // Don't return 304 for HEAD requests
+  if (request.method === 'HEAD') {
```

Comment:
I think this should be before the other conditions (206, 200, 412)

Reply:
Ahh yeah, forgot that we don't support the conditional headers for HEAD requests